### ISSUE TITLE
Wave transformation

### ISSUE THREAD

**Issue by tyrann** [timozattol/wallpaper-generator#4]
I would like to propose a potential improvement to the program. We could include some different types of vertex translation/morphing in order to create different perspective effects on the lattice. Additionally we could change the base lattice shape by triangles or polygons. One of the example I am thinking about is:
![general-desktop](https://cloud.githubusercontent.com/assets/1999872/13456483/a8c82930-e063-11e5-8987-6cbdce459570.png)

One of the first improvement could be to separate the different classes depending on the shapes desired by the user.

**Comment by timozattol** [comment]
Triangles are already supported, see method `initialise` from polylattice:

`def initialise(self, separate_in_triangles=False)`

Setting the second variable to true separates the rectangles into two triangles :)

The idea of vertex transformation is very good, we could create a `Transformation` class that is applied at the end. We still have to decide whether to apply it on vertices before drawing the image, or applying it on the drawn image? Applying it to the drawn image can be neat, because it might result to non-straight edges, we'll have to try! There are probably transformations already included in the library I use for images, `Pillow`.

**Comment by tyrann** [comment]
It's true that the triangles are supported, but only as a feature of a rectangle. Maybe it could be nice to have shapes of different sizes. So we could pass to the some random (or not random) polygon to the initialization function, without requiring the lattice to be completely filled. It's kind of difficult to explain but here is an example.
![geometric_relief_pt2_by_0m3g4_1o](https://cloud.githubusercontent.com/assets/1999872/13459463/1d856746-e074-11e5-9c22-8b620ad9b111.jpg)

![2860136978_f4c06a53fe_m](https://cloud.githubusercontent.com/assets/1999872/13459517/815538d2-e074-11e5-87e5-2bb134e472a4.jpg)

But this may be opened in another issue. For this one let's focus on the different types of transformation on a simple rectangle lattice.

Applying the transformation after drawing could lead to some really interesting effects :+1:

**Comment by timozattol** [comment]
Yep! Kind-of-random polygons that don't intersect each other and fill completely the lattice must be complete madness to design, but your solution of not requiring the lattice to be filled completely is totally doable.

I wanted to fill the lattice completely, but also to have some kind of balance between randomness and regularty, that's why I came up with the vertex mutation solution.

I really like your second image, that would be awesome to find a more "general" algorithm to generate regular compositions of irregular / random objects.

Anyway! We'll open another issue when time comes, let's try those transformations / effects before :+1:
